### PR TITLE
Api server kubelet https

### PR DESCRIPTION
**Pull request by kharyam**
#### Description:

- Implementing rule for api_server_kubelet_https

#### Rationale:

- Implementing check in accordance with the draft CIS spec for OCP 4

### CODE REVIEW

**Comment by evgenyz** [comment]
Seems legit to me

**Comment by evgenyz** [comment]
Meh, I looked into the file the probe tries to get data from. I'm sorry, this change does not make sense as 'config.yaml' contain a string, not a structured object. But, I've incorporated fixes into #6226.

File: applications/openshift/api-server/api_server_kubelet_https_serving_cert/rule.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Verify that the serving-cert for the openshift-apiserver is type
+    <tt>kubernetes.io/tls</tt> and that returned Data includes <tt>tls.crt</tt>
+    and <tt>tls.key</tt>.
+
+ocil: |-
+    Run the following command on the master node(s):
+    <pre>$ oc -n openshift-apiserver describe secret serving-cert</pre>
+    Verify that the <tt>serving-cert</tt> for the openshift-apiserver is type
+    <tt>kubernetes.io/tls</tt> and that returned Data includes <tt>tls.crt</tt>
+    and <tt>tls.key</tt>.
```
Comment (JAORMX) [inline applications/openshift/api-server/api_server_kubelet_https_serving_cert/rule.yml:42]: this one seems to be missing a rule to actually do the check. Is this intentional?

File: applications/openshift/api-server/api_server_kubelet_https_cert/rule.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    Edit the <tt>openshift-kube-apiserver</tt> configmap on the master node(s)
+    and remove the <tt>kubelet-https</tt> parameter if it is set to <tt>false</tt>:
+    <pre>
+    "apiServerArguments":{
+      ...
+      "kubelet-https":[
+        "false"
+      ],
+      ...
+    </pre>
```
Comment (JAORMX) [inline applications/openshift/api-server/api_server_kubelet_https_cert/rule.yml:19]: You might want to change the description to say that you're checking kubelet-client-certificate

File: applications/openshift/api-server/api_server_kubelet_https_key/rule.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    Edit the <tt>openshift-kube-apiserver</tt> configmap on the master node(s)
+    and remove the <tt>kubelet-https</tt> parameter if it is set to <tt>false</tt>:
+    <pre>
+    "apiServerArguments":{
+      ...
+      "kubelet-https":[
+        "false"
+      ],
+      ...
+    </pre>
```
Comment (JAORMX) [inline applications/openshift/api-server/api_server_kubelet_https_key/rule.yml:19]: you might want to change the description to say that you're checking the key

File: applications/openshift/api-server/api_server_kubelet_https_cert/rule.yml (hunk 2 above)
Comment (kharyam) [reply applications/openshift/api-server/api_server_kubelet_https_cert/rule.yml:19]: Thanks, updated

File: applications/openshift/api-server/api_server_kubelet_https_key/rule.yml (hunk 3 above)
Comment (kharyam) [reply applications/openshift/api-server/api_server_kubelet_https_key/rule.yml:19]: Thanks, updated

File: applications/openshift/api-server/api_server_kubelet_https_serving_cert/rule.yml (hunk 1 above)
Comment (kharyam) [reply applications/openshift/api-server/api_server_kubelet_https_serving_cert/rule.yml:42]: This one wants to check a secret, so its just an informational rule to keep the operator permissions as they are.

**Review by JAORMX — APPROVED** [review]

**Pull request merged**